recipes/src/main/resources/quarkus-updates/core/3.3.yaml

```diff
@@ -0,0 +1,33 @@
+#####
+# Adjust properties in application.properties
+#####
+---
+type: specs.openrewrite.org/v1beta/recipe
+name: io.quarkus.updates.core.quarkus33.ApplicationProperties
+applicability:
+  singleSource:
+    - org.openrewrite.FindSourceFiles:
+        filePattern: "**/application*.properties"
+recipeList:
+  - org.openrewrite.properties.ChangePropertyKey:
+      oldPropertyKey: quarkus.hibernate-search-orm.automatic_indexing.synchronization.strategy
+      newPropertyKey: quarkus.hibernate-search-orm.indexing.plan.synchronization.strategy
+  - org.openrewrite.properties.DeleteProperty:
+      propertyKey: quarkus.hibernate-search-orm.automatic_indexing.enable_dirty_check
+
+#####
+# Adjust properties in application.yml/yaml
+#####
+---
+type: specs.openrewrite.org/v1beta/recipe
+name: io.quarkus.updates.core.quarkus33.ApplicationYml
+applicability:
+  singleSource:
+    - org.openrewrite.FindSourceFiles:
+        filePattern: "**/application*.y*ml"
+recipeList:
+  - org.openrewrite.yaml.ChangePropertyKey:
+      oldPropertyKey: quarkus.hibernate-search-orm.automatic_indexing.synchronization.strategy
+      newPropertyKey: quarkus.hibernate-search-orm.indexing.plan.synchronization.strategy
```

Comment:
Note that unfortunately, this won't work very well for YAML as OpenRewrite joins elements with dots which is not supported by SmallRye Config but that's the best we can do.
We have the problem with several other properties, we will have to live with it for now, I think.

There is an open issue here: https://github.com/openrewrite/rewrite/issues/3426.

I tried to see if we could implement our own recipe but I somehow wasn't able to get OpenRewrite to find the constructor when our Java recipes have parameters (and I /think/ we have Lombok configured right).

Reply:
> which is not supported by SmallRye Config

That's the root of the problem IMO. Very annoying as a user. But it's been discussed before and it seems people working on Quarkus config disagree :/

Reply:
See also https://github.com/quarkusio/quarkus/issues/11744